js/ui/camera.js

```diff
@@ -408,13 +409,27 @@ util.extend(Camera.prototype, /** @lends Map.prototype */{
             startZoom = this.getZoom(),
             startBearing = this.getBearing(),
             startPitch = this.getPitch(),
+            startCenter = tr.center,
 
             zoom = 'zoom' in options ? +options.zoom : startZoom,
             bearing = 'bearing' in options ? this._normalizeBearing(options.bearing, startBearing) : startBearing,
             pitch = 'pitch' in options ? +options.pitch : startPitch,
 
             toLngLat,
-            toPoint;
+            toPoint,
+
+            offsetLocation,
+            toLocation,
+            fromLocation,
+            endTransform;
+
+        // Setup a Transform representing the state of the map at the end of the transition
+        endTransform = new Transform();
+        endTransform.center = tr.center;
+        endTransform.resize(tr.width, tr.height);
+        endTransform.zoom = zoom;
+        endTransform.bearing = bearing;
+        endTransform.pitch = 0; // fixes #3119 by pretending the map is not pitched; use endTransform.pitch = pitch to revert to the old behavior
```

Comment:
Oh! I have another idea that'll require much less work:

 - factor `Trasform#pointLocation` into a standalone utility function with the interface 

```js
function pointLocation(
    cameraOptions: CameraOptions,
    mapSize: {width: number, height: number},
    point: Point
)
```